flow/scenarios/highway/scenario.py

```diff
@@ -39,11 +41,36 @@ def __init__(self,
 
         self.length = net_params.additional_params["length"]
         self.lanes = net_params.additional_params["lanes"]
+        self.num_edges = net_params.additional_params.get("num_edges", 1)
 
         super().__init__(name, generator_class, vehicles, net_params,
                          initial_config, traffic_lights)
 
     def specify_edge_starts(self):
         """See parent class."""
-        edgestarts = [("highway", 0)]
+        edgestarts = [("highway_{}".format(i), 0)
+                      for i in range(self.num_edges)]
         return edgestarts
+
+    def gen_custom_start_pos(self, initial_config, num_vehicles, **kwargs):
+        """Generate a user defined set of starting positions.
+        This method is just used for testing.
+
+        Parameters
+        ----------
+        initial_config : InitialConfig type
+            see flow/core/params.py
+        num_vehicles : int
+            number of vehicles to be placed on the network
+        kwargs : dict
+            extra components, usually defined during reset to overwrite initial
+            config parameters
+
+        Returns
+        -------
+        startpositions : list of tuple (float, float)
+            list of start positions [(edge0, pos0), (edge1, pos1), ...]
+        startlanes : list of int
+            list of start lanes
+        """
+        return kwargs["start_positions"], kwargs["start_lanes"]
```

Comment:
do we want this for all scenarios? if so, would be a good idea to create an issue about this

Reply:
It seems like a one-off at the moment? I don't see a compelling case for carefully specifying the initial position of vehicles but happy to be convinced otherwise